### PR TITLE
Optional API key

### PR DESCRIPTION
**Pull request by mitya52**
* optional REFACT_ADMIN_TOKEN env var restricts access to webgui and inference
* if there is no REFACT_ADMIN_TOKEN, server will work for all without any keys or tokens

### CODE REVIEW

File: self_hosting_machinery/webgui/static/login.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    fetch('/login', {
+        method: "POST",
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+            "token": token
+        })
+    }).then(response => response.json()).then(data => {
```
Comment (valaises) [inline self_hosting_machinery/webgui/static/login.js:14]: check if response.ok

File: self_hosting_machinery/webgui/webgui.py [hunk 2]
```diff
@@ -131,10 +168,15 @@ async def init_database():
...
+
+    admin_token = os.environ.get("REFACT_ADMIN_TOKEN", None)
+    session = AdminSession(admin_token) if admin_token is not None else DummySession()
+
     app = WebGUI(
         model_assigner=model_assigner,
         database=database,
         stats_service=stats_service,
+        session=session,
         docs_url=None, redoc_url=None)
```
Comment (valaises) [inline self_hosting_machinery/webgui/webgui.py:180]: not an issue, but is there a reason to hide docs?

**Review by valaises — APPROVED** [review]
seems like it works

**Comment by olegklimov** [comment]
I'm concerted about login.html , where does it come from, how does it look?

**Comment by mitya52** [comment]
> I'm concerted about login.html , where does it come from, how does it look?

it's from enterprise version, login before webgui

**Pull request merged**